xapian-core/tests/harness/backendmanager_honey.cc

```diff
@@ -73,12 +73,62 @@ BackendManagerHoney::do_get_database_path(const vector<string> & files)
     return dbpath;
 }
 
+Xapian::WritableDatabase
+BackendManagerHoney::get_generated_database(const string& name)
+{
+    // creating generated database inside glass cache
+    // to prevent a valid glass db inside honey cache
+    // if testsuite was interrupted
+    string db_path = ".glass/" + name;
+
+    rm_rf(db_path);
+
+    auto flags = Xapian::DB_CREATE|Xapian::DB_BACKEND_GLASS;
+    Xapian::WritableDatabase wdb(db_path, flags);
```

Comment:
This creates the database at the final path in the glass cache directory, so if creation is interrupted a valid DB with the wrong contents may be left in the cache, and if it is a future run could then use for glass tests.  Depending on the contents and testcases which use the DB, this could causing mysterious test failures, or worse tests to pass when they shouldn't.  For those sort of reasons, we carefully create these cached databases at a temporary path and only rename to the final path once they're complete.

This also blows away any existing cached glass version of this database unnecessarily - if there's a cached glass version but no cached honey version we only need to do the conversion step.

It would also be better to use `BackendManagerGlass` to handle the creation of the glass database to convert from - that class already knows how lazily create the database and to safely create it at a temporary path and then rename.  And really `BackendManagerHoney` shouldn't encode knowledge about glass databases.

Reply:
Inside get_database(in backendmanager.cc) the function get_generated_database is being called on a temporary path already(tmp_dbleaf). So the concerns you mentioned won’t be true(or if there’s something else im missing). I agree with the BackendManagerGlass handling the glass part, and would work on its implementation.